project-chip/connectedhomeip
# [IM] Update resource minimas for IM as per spec

#### Problem

Fixes #15831 

#### Change overview
- Check the resource usage on read / subscription requests
- Reject the request with ResourceExhausted status code

#### Testing

Test w/ REPL

Introduced a new flag on build_examples.py (`test`) to force enable some logic.

Introduced a new build flag (`CHIP_CONFIG_IM_FORCE_FABRIC_QUOTA_CHECK=${chip_im_force_fabric_quota_check}`) to limit the resource usage even when the resources are allocated on heap.

- Try to establish as many subscriptions as possible on each fabric, expect success or resource exhausted
- Try to read some attributes with above active subscriptions, expect success
- Verify the numbers of success subscriptions on each fabric are no smaller than 3
